spec/features/return_applications/return_part_payment_applications_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.feature 'When part-payment applications are returned', type: :feature do
+  include Warden::Test::Helpers
+  Warden.test_mode!
+
+  let(:office) { create :office }
+  let(:user) { create :user, office: office }
+
+  let(:application1) { create :application_full_remission, office: office }
+  let!(:evidence1) { create :part_payment, application: application1 }
+  let(:application2) { create :application_full_remission, office: office }
+  let!(:evidence2) { create :part_payment, application: application2 }
+
+  before { login_as user }
+
+  context 'when on home page' do
+
+    before { visit root_path }
+
+    scenario 'shows the applications that are waiting for part-payment' do
+      within '.waiting-for-part_payment' do
+        expect(page).to have_content(application1.reference)
+        expect(page).to have_content(application2.reference)
+      end
+    end
+
+    context 'processing an application' do
+      before { click_link application1.reference }
+      scenario 'shows the application data' do
+        expect(page).to have_content 'Process part-payment'
+        expect(page).to have_content application1.applicant.full_name
+      end
+
+      scenario 'shows "Start now" button' do
+        expect(page).to have_link 'Start now'
+      end
+
+      scenario 'shows "Return application" button' do
+        expect(page).to have_link 'Return application'
+      end
+    end
+
+    scenario 'when returning application' do
+      click_link application1.reference
+      click_link 'Return application'
+      expect(page).to have_content 'Processing complete'
+      expect(page).to have_button 'Finish'
+      click_button 'Finish'
+      expect(page).to have_content 'Start now'
+      within '.waiting-for-part_payment' do
+        expect(page).not_to have_content(application1.reference)
```

Comment:
`not_to` doesn't reliably work with `have_content`. There's a dedicated matcher `have_no_content` for this in Capybara.